### DISCUSSION TITLE
Exactly once reading in flink/spark connector.

### DISCUSSION THREAD

**Discussion by rohankrao** [apache/doris#20677] [A - General / Q&A]
Hi, I wanted to know if the flink/spark source connectors support exactly once reading. If currently not, is it possible to add support for this?

**Comment by yagagagaga** [comment]
Doris is currently unable to do it. You can do it by yourself in your application.

**Reply by rohankrao** [reply]
Could you please explain what changes can I make in the doris flink connector to achieve exactly once reading?

**Reply by yagagagaga** [reply]
It depend on your business. Doris will provide binlog function implementation in the future. You can see it in [Roadmap 2023](https://github.com/apache/doris/issues/16392). If you want to have deeper communication, you can join the [community](https://github.com/apache/doris/discussions/16887).
